src/kafka/ztf.rs

```diff
@@ -315,16 +343,19 @@ impl AlertProducer for ZtfAlertProducer {
             }
             let payload = std::fs::read(path).unwrap();
 
+            let key_string = current_key.to_string();
             let record = FutureRecord::to(&topic_name)
                 .payload(&payload)
-                .key("ztf")
+                .key(&key_string)
                 .timestamp(chrono::Utc::now().timestamp_millis());
 
             producer
                 .send(record, std::time::Duration::from_secs(0))
                 .await
                 .unwrap();
 
+            current_key = (current_key + 1) % nb_partitions as u64;
+
```

Comment:
Have you looked into not specifying a key? My understanding is that kafka uses round-robin by default, which is basically what you're implementing here.